### PR TITLE
OpenAL headers, fix for OSX compile

### PR DESCRIPTION
**Pull request by JohnPaulHarold**
This PR changes how the OpenAL headers are imported, fixing a compile error on OSX. Only tested on OSX.

### CODE REVIEW

**Comment by JayFoxRox** [comment]
Continuation of #249 ?!

Please give a reason and link old PRs / issues if you restart them.
I don't see any reason why it was necessary either.

---

"Thanks" do not belong in a commit title and it's 11 commits for 8 lines of changes.
Please squash up (Commit style is part of the review).

File: .gitignore [hunk 1]
```diff
@@ -4,4 +4,4 @@ documentation/
 *~
 *swp
 *.user*
-
+.DS_Store
```
Comment (JayFoxRox) [inline .gitignore]: unrelated change

**Comment by pcercuei** [comment]
Also don't thank me, I receive an email each time I get mentionned :D

**Comment by JohnPaulHarold** [comment]
I closed the previous PR as, well, it looked like a mess my side, but I thought I'd opened a new PR less commits as I was having a mare with rebase. I'm not sure why that isn't showing.

**Comment by danhedron** [comment]
This is a good change, I'd like to get it merged ASAP.

Could you separate the change to .gitignore into a separate commit to make it clear it's unrelated to fixing OpenAL? You can drop into the IRC channel if you need any help.

**Comment by JohnPaulHarold** [comment]
`.gitignore` change now removed. I will try and make a separate PR for that after this.

**Comment by danhedron** [comment]
Thanks for fixing this.
